packages/amplify-cli/src/amplify-exception-handler.ts

```diff
@@ -202,6 +205,54 @@ const nodeErrorToAmplifyException = (err: NodeJS.ErrnoException): AmplifyExcepti
   );
 };
 
+type CommonExceptionEntry = {
+  messageIndicator: RegExp;
+  amplifyErrorType: AmplifyErrorType;
+  newMessage?: string;
+  resolution: string;
+};
+
+const commonExceptions: CommonExceptionEntry[] = [
+  {
+    messageIndicator: /The provided token has expired./,
+    amplifyErrorType: 'TokenError',
+    newMessage: 'Your AWS access tokens have expired.',
+    resolution: 'Please refresh your tokens.',
+  },
+  {
+    messageIndicator:
+      /(The AWS Access Key Id you provided does not exist in our records.|The request signature we calculated does not match the signature you provided.|The provided token is malformed or otherwise invalid.)/,
+    amplifyErrorType: 'TokenError',
+    newMessage: 'Your AWS tokens are invalid.',
+    resolution: 'Please refresh your tokens.',
+  },
+  {
+    messageIndicator: /is not authorized to perform/,
+    amplifyErrorType: 'PermissionsError',
+    resolution: 'Please update the permissions.',
+  },
+  {
+    messageIndicator: /Rate Exceeded/,
+    amplifyErrorType: 'RateExceededError',
+    resolution: 'Please try again.',
+  },
+];
+
+const commonExceptionsToAmplifyError = (err: AmplifyException): AmplifyException => {
+  if (err.options?.message) {
```

Comment:
nit: negate this condition and early return instead of nesting the whole function in the if